### PR TITLE
docs(dp) sidecar injection with label

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**
We are changing the way how we mark namespaces and pods.

https://github.com/kumahq/kuma/pull/3417

### CODE REVIEW

File: docs/docs/4.4.4/documentation/dps-and-data-model.md [hunk 1]
```diff
@@ -235,6 +235,9 @@ spec:
 
 On Kubernetes the [`Dataplane`](#dataplane-entity) entity is also automatically created for you, and because transparent proxying is being used to communicate between the service and the sidecar proxy, no code changes are required in your applications.
 
+While you can still use annotations instead of labels, we strongly recommend using labels.
```
Comment (parkanzky) [inline docs/docs/4.4.4/documentation/dps-and-data-model.md:238]: Would we say that the annotations are deprecated at this point?

File: docs/docs/4.4.4/documentation/dps-and-data-model.md [hunk 2]
```diff
@@ -294,7 +297,7 @@ spec:
...
+The optimal gateway in Kubernetes mode would be Kong. You can use [Kong for Kubernetes](https://github.com/Kong/kubernetes-ingress-controller) to implement authentication, transformations, and other functionalities across Kubernetes clusters with zero downtime. Using [Kong for Kubernetes](https://github.com/Kong/kubernetes-ingress-controller) with Kuma requires an annotation on every `Service` that you want to pass traffic to [`ingress.kubernetes.io/service-upstream=true`](https://github.com/Kong/kubernetes-ingress-controller/blob/master/docs/references/annotations.md#ingresskubernetesioservice-upstream). This is automatically injected by Kuma for every Kubernetes service that is in a namespace part of the mesh i.e. has `kuma.io/sidecar-injection: enabled` label.
```
Comment (parkanzky) [inline docs/docs/4.4.4/documentation/dps-and-data-model.md:300]: The optimal gateway in Kubernetes mode is Kong

**Review by parkanzky — APPROVED** [review]

File: docs/docs/4.4.4/documentation/dps-and-data-model.md [hunk 3]
```diff
@@ -235,6 +235,9 @@ spec:
 
 On Kubernetes the [`Dataplane`](#dataplane-entity) entity is also automatically created for you, and because transparent proxying is being used to communicate between the service and the sidecar proxy, no code changes are required in your applications.
 
+While you can still use annotations instead of labels, we strongly recommend using labels.
+It's the only way to guarantee that application can only be started with sidecar.
```
Comment (michaelbeaumont) [inline docs/docs/4.4.4/documentation/dps-and-data-model.md:239]:
```suggestion
It's the only way to guarantee that an application can only be started with a sidecar.
```

File: docs/docs/4.4.4/documentation/dps-and-data-model.md (hunk 2 above)
Comment (michaelbeaumont) [inline docs/docs/4.4.4/documentation/dps-and-data-model.md:300]:
```suggestion
The optimal gateway in Kubernetes mode would be Kong. You can use [Kong for Kubernetes](https://github.com/Kong/kubernetes-ingress-controller) to implement authentication, transformations, and other functionalities across Kubernetes clusters with zero downtime. Using [Kong for Kubernetes](https://github.com/Kong/kubernetes-ingress-controller) with Kuma requires the annotation [`ingress.kubernetes.io/service-upstream=true`](https://github.com/Kong/kubernetes-ingress-controller/blob/master/docs/references/annotations.md#ingresskubernetesioservice-upstream) on every `Service` that you want to pass traffic to. This is automatically injected by Kuma for every Kubernetes service that is in a namespace that's part of the mesh i.e. has `kuma.io/sidecar-injection: enabled` label.
```

**Pull request merged**